chapters/request_size.rst

```diff
@@ -0,0 +1,133 @@
+=================
+Request Size
+=================
+
+``request_size`` is the parameter that we could set for the Flow.
+It defines the number of data sent into parallel ``Peas``.
+
+.. contents:: Table of Contents
+    :depth: 2
+
+Request Size
+------------
+Jina defines the ``request_size`` as the parameter at client side. By setting the ``request_size`` in Flow's API
+we could define the number of data sent to each ``Peas``.
+
+Batch Size
+------------
+Batch size, commonly used in machine learning, often refers to the number of data that we feed into
+the model in one iteration.
+
+In Jina, ``batch_size`` is used at ``Driver`` and ``Executor`` level. Where
+we align the usage of machine learning terms for ``batch_size``.
+
+Before you start
+------------------
+Make sure you intall Numpy and latest version of Jina on your local machine.
+
+.. highlight:: bash
+.. code-block:: bash
+
+    pip install numpy
+    pip install -U jina
+
+Implementation
+--------------------
+In order to have a better understanding of the influence of ``request_size`` and how it works, let's take the following
+codes as example.
+
+We first import the necessary modules.
+
+.. highlight:: python
+.. code-block:: python
+
+    import time
+    from typing import Iterator
+
+    import numpy as np
+    from jina import Document
+    from jina.executors.crafters import BaseCrafter
+    from jina.flow import Flow
+
+Then we define a simple crafter which will just forward the data.
+
+.. highlight:: python
+.. code-block:: python
+
+    class SimpleCrafter(BaseCrafter):
+        def craft(self, id, *args, **kwargs):
+            return {'id': id}
+
+
+Next we define the data generator.
+
+.. highlight:: python
+.. code-block:: python
+
+    def random_docs(num_docs, chunks_per_doc=5, embed_dim=10, jitter=1, start_id=0, embedding=True) -> Iterator['Document']:
```

Comment:
This is way too noisy to explain the concept

Reply:
How about just using numpy.random?

Reply:
yes anything where the request size concept shines with easy Documents

Reply:
✅